library/bignum.c

```diff
@@ -606,7 +606,6 @@ int mbedtls_mpi_write_string(const mbedtls_mpi *X, int radix,
     size_t n;
     char *p;
     mbedtls_mpi T;
-    MPI_VALIDATE_RET(buflen == 0 || buf != NULL);
```

Comment:
Do we have a test ensuring this function works with `buflen == 0 && buf == NULL`? (I'm not asking about `buflen !=0 && buf != NULL` because all the positive test will have that.)

From a quick look, I don't think we have such a test. Can you add one? Perhaps a dedicated function like `mpi_zero_length_buffer_is_null()` that would test all functions that are supposed to accept a buffer being NULL when its length is 0. (By "accept" I mean "not crash" - the function might very well return an error though.)